assembl/views/api2/discussion.py

```diff
@@ -740,12 +745,21 @@ def get_visitors(request):
     use_first = asbool(request.GET.get("first", False))
     attribute = "first_visit" if use_first else "last_visit"
     visitors = []
+    from assembl.models import ProfileField
     for st in discussion.agent_status_in_discussion:
         if not getattr(st, attribute, None):
             continue
+        profile_fields = db.query(ProfileField).filter(ProfileField.discussion_id == discussion.id).filter(
+            ProfileField.agent_profile_id == st.agent_profile.id).all()
         data = {"time": getattr(st, attribute),
                 "name": (st.agent_profile.name or '').encode("utf-8"),
                 "email": (st.agent_profile.get_preferred_email() or '').encode("utf-8")}
+
+        for configurable_field in configurable_fields:
```

Comment:
I think that it is not very performant and you should be able to do a query with a join to get all the profile fields for your user that are "linked" to a custom configurable field.

Reply:
indeed
```
profile_fields = db.query(ProfileField).filter(ProfileField.discussion_id == discussion.id).join(ProfileField.configurable_field).filter(
            ProfileField.agent_profile_id == st.agent_profile.id).filter(AbstractConfigurableField.identifier == "CUSTOM").all()
```